roachpb: add proto changes for StoreCapacity, NodeCapacity

wenyihu6 opened the pull request:
**roachpb: add proto changes for StoreCapacity**

This commit adds MMA-related proto changes to StoreCapacity, including the
writes_bytes_per_second field. It is the aggregated sum of average number of
bytes written per second across all replicas on the store. This field is
currently unused and will be populated in future commits. MMA will later use it
to track write bandwidth usage.

Epic: none
Release note: none

---

**roachpb: add proto changes for NodeCapacity**

This commit adds new proto changes to NodeCapacity to track CPU usage and
capacity at the node level. These metrics will be populated in the store
descriptor before it is gossiped. MMA will use them from gossiped store
descriptors to construct store load messages and compute utilization ratios
across multiple dimensions.

Epic: none
Release note: none

wenyihu6 committed:
roachpb: add proto changes for StoreCapacity

This commit adds MMA-related proto changes to StoreCapacity, including the
writes_bytes_per_second field. It is the aggregated sum of average number of
bytes written per second across all replicas on the store. This field is
currently unused and will be populated in future commits. MMA will later use it
to track write bandwidth usage.

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149634)
<!-- Reviewable:end -->

tbg commented:
Mind adding, in this PR, the functionality that correctly populates these in production code (including basic tests)? That way, we don't add "defunct" code to master.

wenyihu6 committed:
roachpb: add proto changes for NodeCapacity

This commit adds new proto changes to NodeCapacity to track CPU usage and
capacity at the node level. These metrics will be populated in the store
descriptor before it is gossiped. MMA will use them from gossiped store
descriptors to construct store load messages and compute utilization ratios
across multiple dimensions.

Epic: none
Release note: none

wenyihu6 reviewed (commented):
Added more comments for clarity. Also included a very basic test in `metadata_test` - I wasn't sure what the best way is for testing proto-only changes, so let me know if this is insufficient.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/149634#-:-OUepeipCd8lb8TFuq8a:b-mvojbu)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

tbg commented:
I believe this is contained in https://github.com/cockroachdb/cockroach/pull/149699.
